$MirOS: ports/net/djbdns/patches/patch-src_dns_name_c,v 1.2 2005/12/14 23:28:31 tg Exp $
--- src/dns_name.c.orig	Sun Feb 11 21:11:23 2001
+++ src/dns_name.c	Sat Mar 26 16:21:22 2016
@@ -2,6 +2,7 @@
 #include "uint16.h"
 #include "byte.h"
 #include "dns.h"
+#include "ip6.h"
 
 static char *q = 0;
 
@@ -40,6 +41,20 @@ int dns_name4(stralloc *out,const char i
   char name[DNS_NAME4_DOMAIN];
 
   dns_name4_domain(name,ip);
+  if (dns_resolve(name,DNS_T_PTR) == -1) return -1;
+  if (dns_name_packet(out,dns_resolve_tx.packet,dns_resolve_tx.packetlen) == -1) return -1;
+  dns_transmit_free(&dns_resolve_tx);
+  dns_domain_free(&q);
+  return 0;
+}
+
+int dns_name6(stralloc *out,const char ip[16])
+{
+  char name[DNS_NAME6_DOMAIN];
+
+  if (ip6_isv4mapped(ip))
+    return dns_name4(out,ip+12);
+  dns_name6_domain(name,ip);
   if (dns_resolve(name,DNS_T_PTR) == -1) return -1;
   if (dns_name_packet(out,dns_resolve_tx.packet,dns_resolve_tx.packetlen) == -1) return -1;
   dns_transmit_free(&dns_resolve_tx);
